src/pages/NewSensor/NewSensorStepThree.js

```diff
@@ -101,10 +101,13 @@ const dispatchToProps = dispatch => {
   const updateLogInterval = value => dispatch(NewSensorActions.updateLogInterval(value));
   const previousTab = () => dispatch(WizardActions.previousTab());
   const exit = () => dispatch(goBack());
-
-  // TODO Use thunks to send commands to the sensor and save to the database.
   const connectToSensor = async () => {
-    await new Promise(r => setTimeout(r, 3000));
+    dispatch(NewSensorActions.updateSensor)
+      .then(dispatch(NewSensorActions.saveSensor))
+      .then(dispatch(gotoSettings()))
```

Comment:
When `withLoadingIndicator` calls this, it awaits `connectToSensor`, but `connectToSensor` doesn't return anything or await anything, so it resolves really quickly, which makes the spinner disappear really quick - think returning the promise or awaiting will ensure it doesn't resolve super quick

Reply:
fixed now, thanks